modules/misc/autoscreenshot.lua

```diff
@@ -0,0 +1,27 @@
+--[[
+    modules\misc\autoscreenshot.lua
+    Automatically takes a screenshot when an achievement is earned
+]]
+local addonName, Loc = ...;
+
+local AutoScreenshotFrame = CreateFrame('Frame', nil, UIParent);
+
+--[[
+    Handles the WoW API Events Registered Below
+
+    @ param Frame $self The Frame that is handling the event 
+    @ param string $event The WoW API Event that has been triggered
+    @ param arg $... The arguments of the Event
+    @ return void
+]]
+local function HandleEvents (self, event, ...)
+    if (event == 'ACHIEVEMENT_EARNED' and PrimaryDB.autoScreenshot) then
+        AutoScreenshotFrame:UnregisterEvent('ACHIEVEMENT_EARNED');
+        C_Timer.After(1, Screenshot);
+        C_Timer.After(4, function() AutoScreenshotFrame:RegisterEvent('ACHIEVEMENT_EARNED') end);

```

Comment:
@kaytotes How do you feel about debouncing the event this way rather than having to hit the onupdate every frame,